src/plasmapy/formulary/laser.py

```diff
@@ -119,3 +136,366 @@ def intensity(
+
+    Notes
+    -----
+    For details, see :cite:t:`ling:2016`\ .
+
+    Examples
+    --------
+    >>> import astropy.units as u
+    >>> wavelength(2.354307546e15 * u.rad / u.s)
+    <Quantity 8e-07 m>
+    """
+    return (2 * np.pi * c / angular_frequency).to(
+        u.m, equivalencies=u.dimensionless_angles()
+    )
+
+
+lambda_ = wavelength
+"""Alias to `~plasmapy.formulary.laser.wavelength`."""
+
+
+@validate_quantities(
+    wavelength={"can_be_negative": False},
+)
+def angular_frequency(wavelength: u.Quantity[u.m]) -> u.Quantity[u.rad / u.s]:
+    r"""
+    Calculate the angular frequency :math:`\omega` of a laser given the
+    the wavelength of the beam :math:`\lambda`.
+
+    The angular frequency of a wave :math:`\omega`
+    can be calculated using:
+
+    .. math::
+        \omega = \frac{2\pi c}{\lambda},
+
+    where :math:`\lambda` is the wavelength of the beam and :math:`c` is the speed of light.
+
+    **Aliases:** `omega_`
+
+    Parameters
+    ----------
+    wavelength : `~astropy.units.Quantity`
+        wavelength of the laser beam (convertible to m).
+
+    Returns
+    -------
+    omega_0 : `~astropy.units.Quantity`
+        The angular frequency of a laser with a given wavelength.
+
+    Notes
+    -----
+    For details, see :cite:t:`ling:2016`\ .
+
+    Examples
+    --------
+    >>> import astropy.units as u
+    >>> angular_frequency(800 * u.nm)
+    <Quantity 2.35456446e+15 rad / s>
+    """
+    return ((c / wavelength) * 2 * np.pi).to(
+        u.rad / u.s, equivalencies=u.dimensionless_angles()
+    )
+
+
+omega_ = angular_frequency
+"""Alias to `~plasmapy.formulary.laser.angular_frequency`."""
+
+
+@validate_quantities(
+    intensity={"can_be_negative": False},
+    wavelength={"can_be_negative": False},
+)
+def normalized_vector_potential(
+    intensity: u.Quantity[u.watt / u.m**2],
+    wavelength: u.Quantity[u.m],
+) -> u.Quantity[u.dimensionless_unscaled]:
+    r"""
+    Calculate the normalized vector potential :math:`a_0` from the intensity :math:`I`
+    and the wavelength :math:`\lambda`.
+
+    The normalized vector potential of a laser
+    is calculated using:
+
+    .. math::
+        a_0= \frac{e \lambda}{m_e \pi} \sqrt{\frac{I} {2 \epsilon_0 c^5}},
+
+    where :math:`e` is the fudamental charge,
+    :math:`m_e` is the mass of an electron,
+    :math:`\epsilon_0` is the permitivity of free space,
+    :math:`c` is the speed of light,
+    :math:`\lambda` is the wavelength and
+    :math:`I_0` is the intensity of the beam.
+
+    **Aliases:** `a0_`
+
+    Parameters
+    ----------
+    intensity : `~astropy.units.Quantity`
+        Intensity of the laser pulse (convertible to W / m\ :sup:`2`).
+    wavelength : `~astropy.units.Quantity`
+        Wavelength of the laser (convertible to m).
+
+    Returns
+    -------
+    a_0 : `~astropy.units.Quantity`
+        The normalized vector potential of a plasma given the intensity and wavelength of
+        a laser.
+
+    Notes
+    -----
```

Comment:
Perhaps add something like..."The normalized vector potential is sometimes known as..."?